src/edu/washington/escience/myria/expression/CounterExpression.java

```diff
@@ -0,0 +1,76 @@
+package edu.washington.escience.myria.expression;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+
+import edu.washington.escience.myria.Type;
+import edu.washington.escience.myria.expression.evaluate.ExpressionOperatorParameter;
+
+/**
+ * Return a sequence of numbers from 0 (inclusive) to operand (exclusive).
+ */
+public class CounterExpression extends UnaryExpression {
```

Comment:
I think a better name might be `SequenceExpression`, as you already said in the javadoc :) "Counter" makes me think of counting something generated from its child, which is not what this expression does.